pkg/cliClient/cliClient_test.go

```diff
@@ -132,11 +132,10 @@ func TestRequestEvaluationPrerunDataSuccess(t *testing.T) {
 		test_requestEvaluationPrerunData_anonymousSuccess(),
 	}
 
-	httpClientMock := mockHTTPClient{}
-
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			body, _ := json.Marshal(tt.mock.response.body)
+			httpClientMock := mockHTTPClient{}
```

Comment:
This change resolves a bug in the tests where the call to Mock.AssertCalled was hitting this edge case. 

> // It can produce a false result when an argument is a pointer type and the underlying value changed after calling the mocked method.

The impact of this was that the tests originally expected a cliClient request to be called when "offlineMode=local" but the assertion was not failing. The test case for offline mode has now been broken into a seperate test to allow for a Mock.AssertNotCalled check to be made which is now the expected behaviour.